src/intelligence_layer/connectors/retrievers/hybrid_qdrant_in_memory_retriever.py

```diff
@@ -0,0 +1,206 @@
+        threshold: The minimum value of the fusion rank score (combined cosine similarity and keyword similarity).
+        retriever_type: The type of retriever to be instantiated.
+            Should be `ASYMMETRIC` for most query-document retrieveal use cases, `SYMMETRIC` is optimized
+            for similar document retrieval.
+        distance_metric: The distance metric to be used for vector comparison.
+        sparse_model_name: The name of the sparse embedding model from `fastemebed` to be used.
+
+    Example:
+        >>> from intelligence_layer.connectors import LimitedConcurrencyClient, Document, HybridQdrantInMemoryRetriever
+        >>> client = LimitedConcurrencyClient.from_env()
+        >>> documents = [Document(text=t) for t in ["I do not like rain.", "Summer is warm.", "We are so back."]]
+        >>> retriever = HybridQdrantInMemoryRetriever(documents, 5, client=client)
+        >>> query = "Do you like summer?"
+        >>> documents = retriever.get_relevant_documents_with_scores(query)
+    """
+
+    MAX_WORKERS = 10
+
+    def __init__(
+        self,
+        documents: Sequence[Document],
+        k: int,
+        client: AlephAlphaClientProtocol | None = None,
+        threshold: float = 0.0,
+        retriever_type: RetrieverType = RetrieverType.ASYMMETRIC,
+        distance_metric: Distance = Distance.COSINE,
+        sparse_model_name: str = "bm25",
+    ) -> None:
+        self._client = client or LimitedConcurrencyClient.from_env()
+        self._search_client = QdrantClient(":memory:")
+        self._collection_name = "in_memory_collection"
+        self._k = k
+        self._threshold = threshold
+        self._query_representation, self._document_representation = retriever_type.value
+        self._distance_metric = distance_metric
+
+        self._search_client.set_sparse_model(sparse_model_name)
+        self._sparse_vector_field_name = "text-sparse"
+        self._dense_vector_field_name = "text-dense"
+
+        if self._search_client.collection_exists(collection_name=self._collection_name):
+            self._search_client.delete_collection(
+                collection_name=self._collection_name,
+            )
+
+        self._search_client.create_collection(
+            collection_name=self._collection_name,
+            vectors_config=VectorParams(size=128, distance=self._distance_metric),
+            sparse_vectors_config=self._search_client.get_fastembed_sparse_vector_params(),
+        )
+
+        self._add_texts_to_memory(documents)
+
+    def _sparse_embed_query(self, query: str) -> models.SparseVector:
+        if self._search_client.sparse_embedding_model_name is None:
+            raise ValueError("Sparse embedding model is not set!")
+        sparse_embedder = self._search_client._get_or_init_sparse_model(
+            model_name=self._search_client.sparse_embedding_model_name
+        )
+
+        sparse_vector = next(sparse_embedder.query_embed(query=query))
```

Comment:
why are we getting the first element only here?

Reply:
The `query_embed` method works in batch mode by default and we are calling it with a single example/input.